th2_data_services/provider/v5/commands/grpc.py

```diff
@@ -11,43 +11,132 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
-
 from datetime import datetime
 from typing import List, Iterable
 
-from th2_grpc_data_provider.data_provider_pb2 import (
+from grpc._channel import _InactiveRpcError
+from th2_grpc_data_provider.data_provider_template_pb2 import (
     EventData,
     StreamResponse,
     MessageData,
 )
 
 from th2_data_services import Filter
-from th2_data_services.provider.v5.command import IGRPCProvider5Command
+from th2_data_services.provider.command import IGRPCProviderAdaptableCommand
+from th2_data_services.provider.v5.command import (
+    IGRPCProvider5Command,
+    IGRPCProvider5EventCommand,
+    IGRPCProvider5MessageCommand,
+)
 from th2_data_services.provider.v5.data_source.grpc import GRPCProvider5DataSource
 from th2_data_services.provider.v5.provider_api import GRPCProvider5API
+from th2_data_services.provider.v5.stub_builder import provider5_event_stub_builder, provider5_message_stub_builder
+
+
+class GetEventByIdGRPCObject(IGRPCProvider5Command, IGRPCProviderAdaptableCommand):
+    """A Class-Command for request to rpt-data-provider.
 
+    It retrieves the event by id as GRPC object.
+    """
 
-class GetEventById(IGRPCProvider5Command):
     def __init__(self, id: str):
+        """
+        Args:
+            id: Event id.
+
+        """
+        super().__init__()
         self._id = id
 
     def handle(self, data_source: GRPCProvider5DataSource) -> EventData:
         api: GRPCProvider5API = data_source.source_api
-        response = api.get_event(self._id)
-        return response
+        event = api.get_event(self._id)
+
+        event = self._handle_adapters(event) if self._workflow else event
+        return event
+
+
+class GetEventById(IGRPCProvider5EventCommand, IGRPCProviderAdaptableCommand):
+    """A Class-Command for request to rpt-data-provider.
+
+    It retrieves the event by id.
+    """
+
+    def __init__(self, id: str):
+        """
+        Args:
+            id: Event id.
+
+        """
+        super().__init__()
+        self._id = id
+        self._stub_status = False
+
+    def handle(self, data_source: GRPCProvider5DataSource) -> dict:
+        event = {provider5_event_stub_builder.event_fields.EVENT_ID: self._id}
```

Comment:
We should use `data_source.event_struct.EVENT_ID` for this aim.
**DataSource** and the constructor should provide all necessary data for the handler.